arviz/plots/plot_utils.py

```diff
@@ -247,3 +247,29 @@ def xarray_to_nparray(data, *, var_names=None, combined=True):
         unpacked_var_names.append(make_label(var_name, selection))
 
     return unpacked_var_names, np.array(unpacked_data)
+
+
+def get_coords(data, coords):
+    """Subselects xarray dataset object to provided coords. Raises exception if fails
+
+    Raises
+    ------
+    ValueError
+        If coords name are not available in data
+
+    KeyError
+        If coords dims are not available in data
+
+    Returns
+    -------
+    data: xarray
+        xarray.Dataset object
+    """
+    try:
+        return data.sel(**coords)
+
+    except ValueError as err:
+        raise ValueError("Verify coords keys. {}".format(err))
+
+    except KeyError as err:
```

Comment:
...are these right? A `KeyError` gets thrown for bad `.values` and a `ValueError` for bad `keys`?

Reply:
Maybe not. I'll double check. Thanks for double checking

Reply:
So yes,
Having a wrong key in coords dictionary raises a ValueError in xarray, and having a value in the coords dictionary raises a KeyError.

I wrote tests but just double checked by hand as well

Ill make the exceptions more clear